### ISSUE TITLE
UnsupportedMediaTypeException when testing

### ISSUE THREAD

**Issue by Slessi** [Neko-Design/sentry-msteams#11]
On some (not all) of our sentry projects, we see the following error when testing:

```
Test Results:
["System.Net.Http.UnsupportedMediaTypeException: No MediaTypeFormatter is available to read an object of type 'JToken' from conten","t with media type 'application/octet-stream'."]
```

(webhook URL omitted)

![image](https://user-images.githubusercontent.com/3223500/88372824-f35d7c00-cd96-11ea-846f-83ff1d55bd7a.png)

**Comment by Neko-Design** [comment]
Thanks for reporting the issue @Slessi, can i get a few details about the instances that are showing this error such as version and platform and I'll attempt to recreate the problem on my end.

**Comment by Neko-Design** [comment]
Righty, looks like around mid-March the Sentry team refactored the base API client and changed the way it handles reading the response. I'll need to change they way we post the message out to take advantage of the new model, but I suspect this will be a breaking change for any older versions of Sentry using this plugin, which I suspect includes the couple of instances you mention where it's still working.

I'll add a notice on the readme once I've made the change and this will be our first major version bump to indicate that this is going to be non-backwards-compatible.

**Comment by Neko-Design** [comment]
Hey @Slessi, I've deleted my previous comment as I've done a bit more digging about the API change. The old API is still in place and handles the current response from O365 correctly. The library mentioned in the error you're seeing, and indeed the message itself is actually the body of the response received when you fired the webhook, sent from 0365 itself.

The library mentioned is part of the .Net core libs used I would imagine in the receiver backend, which seemed to be having an issue at the time you sent the notification. Can you please give it another go and if you still see this error on one of the instances, send over the console logs from the worker container or instance and we can raise it to Microsoft directly.

Link to the library documentation - https://docs.microsoft.com/en-us/dotnet/api/system.net.http?view=dotnet-plat-ext-3.1

**Comment by Neko-Design** [comment]
Additionally if you'd like to validate this yourself, once you have an instance that reliably throws the error you can send the request directly from the CLI using cURL:

`curl -H 'Content-Type: application/json' -d '{"text": "Webhook POST from CLI"}' https://outlook.office.com/webhook/example-12345-67890`

**Comment by Slessi** [comment]
When I run your command from CLI using the webhook URL for the project that the test button works for, I receive a `1` as in the UI and it works.

When I run the same command using the webhook URL for the project which does *not* work, I get:

> Webhook Bad Request - Null or empty event

Once it became apparent that the webhook URL is the issue, I realised the webhooks I created in teams were:
- Webhook connector for the one that is working
- Sentry connector (the official plugin) for the that is *not* working

So it seems this library only works for the webhook connectoor.
Closing as there doesn't seem to be any actual problem here, maybe ticket will help someone else out at least.
